File: Auth controller.java
public class AuthController {
    private UserModel model;
    private AuthView view;

    public AuthController(UserModel model, AuthView view) {
        this.model = model;
        this.view = view;
    }

    public void handleRegister() {
        String[] data = view.getRegisterData();
        if(data[0].isEmpty() || data[1].isEmpty() || data[2].isEmpty()){
            view.showMessage("Please fill all fields");
            return;
        }
        boolean result = model.register(data[0], data[1], data[2]);
        if(result) {
            view.showMessage("Registration Successful!");
        }
    }

    public void handleLogin() {
        String[] data = view.getLoginData();
        boolean result = model.login(data[0], data[1]);
        if(result) {
            view.showMessage("Login Successful!");
        } else {
            view.showMessage("Login Failed!");
        }
    }
}

File: Auth controller.js
// Controller connects Model and View
class AuthController {
  constructor(model, view){ this.model=model; this.view=view; }
  handleRegister(){
    const d = this.view.getRegisterData();
    if(!d.name||!d.email||!d.password) return this.view.showMessage('regMsg','All fields required',true);
    const r = this.model.register(d.name,d.email,d.password);
    this.view.showMessage('regMsg',r.message,!r.success);
    if(r.success) setTimeout(()=>this.view.showTab('login'),1000);
  }
  handleLogin(){
    const d = this.view.getLoginData();
    const r = this.model.login(d.email,d.password);
    if(r.success){ localStorage.setItem('currentUser',JSON.stringify(r.user)); Router.navigate(this.view,'dashboard',r.user); }
    else this.view.showMessage('loginMsg',r.message,true);
  }
  handleLogout(){ localStorage.removeItem('currentUser'); this.view.hideDashboard(); }
}
